lib/sqlalchemy/dialects/postgresql/ranges.py

```diff
@@ -80,6 +80,85 @@ def __init__(
     def __bool__(self) -> bool:
         return self.empty
 
+    def contains_value(self, value: _T) -> bool:
```

Comment:
so I forgot that we're SQLAlchemy, and we make Python operators symmetric with the SQL operators.   Assuming this is equivalent to the "@>" opeator, this op should be called [contains](https://github.com/sqlalchemy/sqlalchemy/blob/b993bc06fe6536c2d08b344f77b17e5cffa93587/lib/sqlalchemy/dialects/postgresql/ranges.py#L212).

We could in theory add Python versions of all the operators below:

https://github.com/sqlalchemy/sqlalchemy/blob/b993bc06fe6536c2d08b344f77b17e5cffa93587/lib/sqlalchemy/dialects/postgresql/ranges.py#L205-L273

Reply:
It seems plausible, but just to be sure I got your suggestion right: do you mean that I should _collapse_ the two new methods `contains_value()` and `issuperset()` into a single `contains()` one, and then discriminating the actual impl on the type of the first argument? And then rename `issubset()` with `contained_by()`?

Reply:
Hm I see your point, is that what PG's "@>" operator does?  That is, for a daterange, you can hand it an individual date, or another range?   Odd that PG would do that, however, yes it's usually best if Python and SQL interfaces look the same.  it allows one to write code that works in both contexts transparently.    I wrote a blog post about this many years ago at https://techspot.zzzeek.org/2011/10/21/hybrids-and-value-agnostic-types/ - see the "Amount" object, which works in both Python client-side and SQL server-side contexts.

Reply:
cc @CaselIT if you have any reservations about this.

Reply:
> is that what PG's "@>" operator does? That is, for a daterange, you can hand it an individual date, or another range?

Yes, you can see that behaviour in action in the two tests I wrote (although I used the `<@` operator actually), giving a single date for the `contains_value()` and a range for the `issubset()`.

> Odd that PG would do that

Why? Many operators are _overloaded_, like `+` that may sum two integers, or two floats, or a date and an interval...

Reply:
I will wait for Federico's POV.

I think that if we go that way, we could have, say:
```
  def _contains_value(self, scalar_value):
    # current impl of contains_value()

  def _contains_range(self, range_value):
    # current impl of issuperset()

  def contains(self, value):
    if isinstance(value, Range):
      return self._contains_range(value)
    else
      return self._contains_value(value)
```

How does that sound? ... I would need some help in _typification_ though :-)

Reply:
yeah it would work that way, with the branching aspect of it.

Reply:
I know next to nothing about typing, but wouldn't mypy complain about calling either methods, depending on the type of `value`? Shall I use a `cast()` or something?

Reply:
for typing, we can help with that part, but short answer is there would be two `@overload` versions for the method.   no need to get into that we can do it if you dont want to.